invokeai/frontend/web/src/app/NodeAPITest.tsx

```diff
@@ -0,0 +1,200 @@
+    (state: RootState) => state.api
+  );
+
+  const handleCreateSession = () => {
+    dispatch(
+      createSession({
+        requestBody: {
+          nodes: {
+            a: {
+              id: 'a',
+              type: 'txt2img',
+              prompt: 'pizza',
+              steps: 50,
+              seed: 123,
+            },
+            b: {
+              id: 'b',
+              type: 'img2img',
+              prompt: 'dog',
+              steps: 50,
+              seed: 123,
+              strength: 0.9,
+            },
+            c: {
+              id: 'c',
+              type: 'img2img',
+              prompt: 'cat',
+              steps: 50,
+              seed: 123,
+              strength: 0.9,
+            },
+          },
+          edges: [
+            {
+              source: { node_id: 'a', field: 'image' },
+              destination: { node_id: 'b', field: 'image' },
+            },
+            {
+              source: { node_id: 'b', field: 'image' },
+              destination: { node_id: 'c', field: 'image' },
+            },
+          ],
+        },
+      })
+    );
+  };
+
+  const handleInvokeSession = () => {
+    dispatch(invokeSession());
+  };
+
+  const handleCancelProcessing = () => {
+    dispatch(cancelProcessing());
+  };
+
+  useEffect(() => {
+    if (!sessionId) {
+      return;
+    }
+
+    // set up socket.io listeners
+
+    // TODO: suppose this should be handled in the socket.io middleware
+    // TODO: write types for the socket.io payloads, haven't found a generator for them yet...
+
+    // subscribe to the current session
+    socket.emit('subscribe', { session: sessionId });
+    console.log('subscribe', { session: sessionId });
+
+    // received on each generation step
+    socket.on('generator_progress', (data: GeneratorProgress) => {
+      console.log('generator_progress', data);
+      dispatch(setProgress(data.step / data.total_steps));
+      dispatch(setProgressImage(data.progress_image));
+    });
+
+    // received after invokeSession called
+    socket.on('invocation_started', (data) => {
+      console.log('invocation_started', data);
+      dispatch(setStatus(STATUS.busy));
+    });
+
+    // received when generation complete
+    socket.on('invocation_complete', (data) => {
+      // for now, just unsubscribe from the session when we finish a generation
+      // in the future we will want to continue building the graph and executing etc
+      console.log('invocation_complete', data);
+      dispatch(setProgress(null));
+      // dispatch(setSessionId(null));
+      dispatch(setStatus(STATUS.idle));
+
+      // think this gets a blob...
+      // dispatch(
+      //   getImage({
+      //     imageType: data.result.image.image_type,
+      //     imageName: data.result.image.image_name,
+      //   })
+      // );
+    });
+
+    // not sure when we get this?
+    socket.on('session_complete', (data) => {
```

Comment:
I asked kyle the difference between `invocation_complete` and `graph_execution_state_complete`. 

> invocation is for a single node (I think I need to go through the code and rename them all to use "node"). Session is for the entire graph completing execution (at least everything that's in the graph).